husky-ws/docker/docker-compose.yaml

```diff
@@ -0,0 +1,46 @@
+version: '3'
+services:
+  husky-ws:
+    build: .
+    image: yuzhong1214/husky-ws
+    container_name: husky-ws
+    stdin_open: true
+    tty: true
+    privileged: true
+    command: /bin/bash
+    network_mode: host
+    working_dir: /home/ros2-agv-essentials/husky-ws
+    environment:
+      - DISPLAY=${DISPLAY}
+      # Set ros2 environment variables.
+      # References:
+      # - https://docs.ros.org/en/humble/Concepts/Intermediate/About-Domain-ID.html
+      # - https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Configuring-ROS2-Environment.html
+      - ROS_LOCALHOST_ONLY=1
+      - ROS_DOMAIN_ID=42
+      - ROS2_WS=/home/ros2-agv-essentials/husky-ws
+    # Turn on gpu access
+    # Reference : https://docs.docker.com/compose/gpu-support/
+    deploy:
+      resources:
+        reservations:
+          devices:
+            - driver: nvidia
+              count: all
+              capabilities: [ gpu ]
+    volumes:
+      # Mount local timezone into container. ( Readonly )
+      # Reference: https://stackoverflow.com/questions/57607381/how-do-i-change-timezone-in-a-docker-container
+      - /etc/timezone:/etc/timezone:ro
+      - /etc/localtime:/etc/localtime:ro
+      # Mount X11 server
+      - /tmp/.X11-unix:/tmp/.X11-unix
+      # Direct Rendering Infrastructure
+      - /dev/dri:/dev/dri
+      # Mount sound card to prevent Gazebo warning.
+      - /dev/snd:/dev/snd
+      # Mount Gazebo models directory to reuse models downloaded during first launch.
+      # Reference: https://answers.ros.org/question/365658
+      - ./cache/.gazebo:/home/user/.gazebo
+      # Mount workspace
+      - ../..:/home/ros2-agv-essentials
```

Comment:
The `.git` folder is under the `ros2-agv-essentials` folder, if we only mount the working directory into the container, which is `husky_ws`, we cannot access git inside the container.  
  
I think it will be better to mount the outer directory. (It is not necessary, however, it will be convenient)

Reply:
No problem. I was wondering if we should also update the template workspace to mount the outer directory? (in another PR of course)

Reply:
I will open a PR to fix it later.  
Also, I found a mistake in template_ws, I will fix it together.   

https://github.com/j3soon/ros2-agv-essentials/blob/dba4082f5be7422ad8dd8f4ae3ef8f7544b4db0c/template_ws/.gitignore#L1

Here should be `docker/cache`.